neurolib/optimize/evolution/evolution.py

```diff
@@ -813,36 +738,59 @@ def loadEvolution(self, fname):
         evolution.paramInterval = evolution.parameterSpace.named_tuple
-        # # we reinitialize the evolution with the arguments of the saved evolution
-        # evolution.__init__(
-        #     evalFunction=evolution.evalFunction,
-        #     parameterSpace=pars,
-        #     weightList=evolution.weightList,
-        #     model=evolution.model,
-        #     ncores=evolution.ncores,
-        #     POP_INIT_SIZE=evolution.POP_INIT_SIZE,
-        #     POP_SIZE=evolution.POP_SIZE,
-        #     NGEN=evolution.NGEN,
-        #     matingOperator=evolution.matingOperator,
-        #     MATE_P=evolution.MATE_P,
-        #     mutationOperator=evolution.mutationOperator,
-        #     MUTATE_P=evolution.MUTATE_P,
-        #     selectionOperator=evolution.selectionOperator,
-        #     SELECT_P=evolution.SELECT_P,
-        #     parentSelectionOperator=evolution.parentSelectionOperator,
-        #     PARENT_SELECT_P=evolution.PARENT_SELECT_P,
-        #     individualGenerator=evolution.individualGenerator,
-        # )
-
         return evolution
 
+    def _outputToDf(self, pop, df):
+        """Loads outputs dictionary from evolution from the .outputs attribute
+        and writes data into a dataframe.
+
+        :param pop: Population of which to get outputs from.
+        :type pop: list
+        :param df: Dataframe to which outputs are written
+        :type df: pandas.core.frame.DataFrame
+        :return: Dataframe with outputs
+        :rtype: pandas.core.frame.DataFrame
+        """
+        assert len(pop) == len(df), "Dataframe and population do not have same length."
+        # load outputs into dataframe
+        for i, p in enumerate(pop):
+            if hasattr(p, "outputs"):
+                for key, value in p.outputs.items():
+                    # only save floats, ints and arrays
+                    if isinstance(value, (float, int, np.ndarray)):
+                        # save 1-dim arrays
+                        if isinstance(value, np.ndarray):
+                            # to save a numpy array, convert column to object type
+                            if key not in df:
+                                df[key] = None
+                            df[key] = df[key].astype(object)
+                            df.at[i, key] = value
+        return df
+
+    def _dropDuplicatesFromDf(self, df):
+        """Drops duplicates from dfEvolution dataframe.
+        Tries vanilla drop_duplicates, which fails if the Dataframe contains
+        data objects like numpy.arrays. Tries to drop via key "id" if it fails.
+
+        :param df: Input dataframe with duplicates to drop
+        :type df: pandas.core.frame.DataFrame
+        :return: Dataframe without duplicates
+        :rtype: pandas.core.frame.DataFrame
+        """
+        try:
+            df = df.drop_duplicates()
+        except:
+            logging.info('Failed to drop_duplicates() without column name. Trying by column "id".')
```

Comment:
```suggestion
            logging.warning('Failed to drop_duplicates() without column name. Trying by column "id".')
```
?

Reply:
I think this one can stay info, since it's not really worse than doing by "id", just a special case.